tests/runtime/ray/test_dag.py

```diff
@@ -36,81 +36,180 @@ def client():
-        (_client.WorkflowStatus.FAILED, TEST_TIME, None, State.FAILED),
-        # A task that finished before other tasks failed.
-        (_client.WorkflowStatus.FAILED, TEST_TIME, TEST_TIME, State.SUCCEEDED),
-        # Resumable: something's happened to the workflow. Ray allows to
-        # restart it. We don't support that yet -> we say it's failed.
-        # We don't have API for cancelling a wf, either. If a wf is in this
-        # state, something bad happened.
-        (_client.WorkflowStatus.RESUMABLE, None, None, State.FAILED),
-        (_client.WorkflowStatus.RESUMABLE, TEST_TIME, None, State.FAILED),
-        (_client.WorkflowStatus.RESUMABLE, TEST_TIME, TEST_TIME, State.FAILED),
-        # Cancelled: a user stopped the workflow permanently.
-        (_client.WorkflowStatus.CANCELED, None, None, State.TERMINATED),
-        (_client.WorkflowStatus.CANCELED, TEST_TIME, None, State.TERMINATED),
-        (_client.WorkflowStatus.CANCELED, TEST_TIME, TEST_TIME, State.TERMINATED),
+        (_client.WorkflowStatus.CANCELED, start, end, failed, State.TERMINATED)
+        for start in [None, TEST_TIME]
+        for end in [None, TEST_TIME]
+        for failed in [None, False, True]
+    ]
+    # Resumable: something's happened to the workflow. Ray allows the user to restart
+    # it. We don't support that yet -> we say it's failed. We don't have API for
+    # cancelling a wf, either. If a wf is in this state, something bad happened.
+    + [
+        (_client.WorkflowStatus.RESUMABLE, start, end, failed, State.FAILED)
+        for start in [None, TEST_TIME]
+        for end in [None, TEST_TIME]
+        for failed in [None, False, True]
+    ]
+    # Any other workflow status, plus a failed flag: the task has failed, regardless of
+    # the overall workflow status, and should be reported as such.
+    + [
+        (status, start, end, True, State.FAILED)
+        for status in [
+            _client.WorkflowStatus.SUCCESSFUL,
+            _client.WorkflowStatus.RUNNING,
+            _client.WorkflowStatus.FAILED,
+        ]
+        for start in [None, TEST_TIME]
+        for end in [None, TEST_TIME]
+    ]
+    # Any other workflow status, no failed flag, no start time: The task is waiting to
+    # run.
+    + [
+        (status, None, None, False, State.WAITING)
+        for status in [
+            _client.WorkflowStatus.SUCCESSFUL,
+            _client.WorkflowStatus.RUNNING,
+            _client.WorkflowStatus.FAILED,
+        ]
+    ]
+    # Any other workflow status, no failed flag, a start time but no end time: The task
+    # is in progress
+    + [
+        (status, TEST_TIME, None, False, State.RUNNING)
+        for status in [
+            _client.WorkflowStatus.SUCCESSFUL,
+            _client.WorkflowStatus.RUNNING,
+            _client.WorkflowStatus.FAILED,
+        ]
+    ]
+    # Any other workflow status, no failed flag, start time and end time: The task
+    # finished and didn't fail.
```

Comment:
I want with option c since the other tests already check for both False and None: [e9845d9](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/317/commits/e9845d9be6ad75cbf3dbc4702aedcc9cdafc81e9)